src/emmetHelper.ts

```diff
@@ -4,28 +4,33 @@
  *--------------------------------------------------------------------------------------------*/
 
 
-import { Position, Range, CompletionItem, CompletionList, TextEdit, InsertTextFormat, CompletionItemKind } from 'vscode-languageserver-types'
-import { TextDocument } from 'vscode-languageserver-textdocument';
 import * as JSONC from 'jsonc-parser';
-import { cssData, htmlData } from './data';
-import { URI } from 'vscode-uri';
-import { FileService, joinPath, isAbsolutePath, FileType, FileStat } from './fileService';
 import { TextDecoder } from 'util';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import { CompletionItem, CompletionItemKind, CompletionList, InsertTextFormat, Position, Range, TextEdit } from 'vscode-languageserver-types';
+import { URI } from 'vscode-uri';
+import { cssData, htmlData } from './data';
+import { FileService, FileStat, FileType, isAbsolutePath, joinPath } from './fileService';
 
 import expand, { Config, extract, ExtractOptions, MarkupAbbreviation, Options, parseMarkup, parseStylesheet, resolveConfig, stringifyMarkup, stringifyStylesheet, StylesheetAbbreviation, SyntaxType, UserConfig } from 'emmet';
 import { parseSnippets, SnippetsMap, syntaxes } from './configCompat';
 
-import * as nls from 'vscode-nls';
-
-const localize = nls.loadMessageBundle();
-
 // /* workaround for webpack issue: https://github.com/webpack/webpack/issues/5756
 //  @emmetio/extract-abbreviation has a cjs that uses a default export
 // */
 // const extract = typeof _extractAbbreviation === 'function' ? _extractAbbreviation : _extractAbbreviation.default;
 
+export { FileService, FileType, FileStat };
 
-export { FileService, FileType, FileStat }
+let l10n: { t: (message: string) => string };
+try {
+	l10n = require('vscode').l10n;
+} catch {
+	// Fallback to the identity function.
+	l10n = {
+		t: (message: string) => message
+	};
```

Comment:
I'm curious why this fallback is necessary--if it's to deal with older VS Code versions which don't implement `l10n`, perhaps we could declare this requirement via the VS Code `"engine": "^1.74.0"`, unless Emmet has a lot of users who don't update to the latest VS Code for some reason?

Reply:
Since this repo serves as a helper package for other extensions and [LSP implementations](https://www.npmjs.com/browse/depended/@vscode/emmet-helper), I'm unsure whether the `vscode` module is always available. This fallback is also a cheap way to get the unit tests running locally without needing `vscode-test`.